src/components/CurrentUser/CurrentUser.jsx

```diff
@@ -1,12 +1,80 @@
 import React from 'react';
 import './CurrentUser.scss';
+import PropTypes from 'prop-types';
+import { getUser } from '../../api';
 
-export const CurrentUser = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+export class CurrentUser extends React.Component {
+  state = {
+    selectedUser: null,
+    isUserSelected: false,
+  }
 
-    <h3 className="CurrentUser__name">Ervin Howell</h3>
-    <p className="CurrentUser__email">Shanna@melissa.tv</p>
-    <p className="CurrentUser__phone">010-692-6593 x09125</p>
-  </div>
-);
+  componentDidMount() {
+    getUser(this.props.userId)
+      .then(user => this.setState({
+        selectedUser: user,
+        isUserSelected: true,
+      }));
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.userId !== this.props.userId) {
+      getUser(this.props.userId)
+        .then(user => this.setState({
+          selectedUser: user,
+          isUserSelected: true,
+        }));
+    }
+  }
+
+  clearUser = () => {
+    this.props.clearId();
+
+    this.setState({
+      selectedUser: null,
+      isUserSelected: false,
+    });
+  }
+
+  render() {
+    const { selectedUser, isUserSelected } = this.state;
+    const pageIsLoading = (!isUserSelected
+      ? <span>User is not selected</span>
+      : (
+        <div className="CurrentUser">
+          <h2 className="CurrentUser__title">
+            <span>
+              User&nbsp;#
+              {selectedUser.id}
+            </span>
+          </h2>
+          <h3 className="CurrentUser__name">
+            {selectedUser.name}
+          </h3>
+          <p className="CurrentUser__email">{selectedUser.email}</p>
+          <p className="CurrentUser__phone">{selectedUser.phone}</p>
+          <button
+            className="
+            TodoList__user-button
+            button
+          "
```

Comment:
```suggestion
            className="TodoList__user-button button"
```
2 класса можно писать в 1 строку